CONVENTIONS.md

```diff
@@ -63,6 +64,14 @@ apply to instance fields, as it's useful to see everything that
 constitutes the state of an object in one place. This is also why
 properties should appear alongside instance fields.
 
+Optional parameters
+---
+
+Optional parameters should be used very sparingly outside the guidelines
+listed here to avoid overload explosion. In particular, any suggested use of
+a default argument value other than the default value for the parameter type is strongly
+discouraged and should be discussed before implementation.
```

Comment:
Mention the fact that the value used in optional parameters become part of the API and therefore changing it is a breaking change?